sqlite/sqlite.go

```diff
@@ -0,0 +1,144 @@
+package sqlite
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strconv"
+	"sync"
+
+	// sqlite3 driver
+	_ "github.com/mattn/go-sqlite3"
+
+	"go.uber.org/zap"
+)
+
+const (
+	DefaultFilename = "influxd.sqlite"
+	InmemPath       = ":memory:"
+)
+
+// SqlStore is a wrapper around the db and provides basic functionality for maintaining the db
+// including flushing the data from the db during end-to-end testing.
+type SqlStore struct {
+	mu  sync.Mutex
+	db  *sql.DB
+	log *zap.Logger
+}
+
+func NewSqlStore(path string, log *zap.Logger) (*SqlStore, error) {
+	db, err := sql.Open("sqlite3", path)
+	if err != nil {
+		return nil, err
+	}
+	log.Info("Resources opened", zap.String("path", path))
+
+	// If using an in-memory database, don't allow more than 1 connection. Each connection
+	// is given a "new" database. We can't use a shared cache in-memory database because
+	// parallel tests that run multiple launchers in the same process will have issues doing
+	// concurrent writes to the database. See: https://sqlite.org/inmemorydb.html
+	if path == InmemPath {
+		db.SetMaxOpenConns(1)
+	}
+
+	return &SqlStore{
+		db:  db,
+		log: log,
+	}, nil
+}
+
+// Close the connection to the sqlite database
+func (s *SqlStore) Close() error {
+	err := s.db.Close()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Flush deletes all records for all tables in the database.
```

Comment:
I wouldn't call this operation `Flush`, I'd call it `Truncate`. Flush to me means ensuring that any data cached in-memory is forced out to disk.

Reply:
I agree about the terminology......but the `Flush` name is used in lots of places, on the bolt store, on the engine, even in the API at the `/debug/flush` endpoint. It's all used for clearing the store of data for end-to-end testing. I'd have to change quite a bit of stuff to make it consistent - even the API and the UI e2e tests and I suppose also the same endpoint that cloud uses for UI e2e testing. So I'm going to advocate for leaving this as-is unless you think it's a deal-breaker.

Reply:
😱😱😱

That's some very unfortunate naming... but I'd agree consistency is probably better at this point.